### PR TITLE
feat: find executable filename to address #16

### PR DESCRIPTION
**Pull request by tsutterley**
should address #4 for finding relative paths

### CODE REVIEW

File: sphinxarg/ext.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 import os
 import sys
 from argparse import ArgumentParser
+from distutils.spawn import find_executable
```
Comment (ashb) [inline sphinxarg/ext.py:4]: We can't use distutils, it's going away in Py 3.12:

> [distutils](https://docs.python.org/3/library/distutils.html#module-distutils) is deprecated with removal planned for Python 3.12. See the [What’s New](https://docs.python.org/3/whatsnew/3.10.html#distutils-deprecated) entry for more information.

**Review by ashb — CHANGES_REQUESTED** [review]
Please ping/mention me when you've made the changes.

(Sorry for not noticing this earlier, I didn't have notifications on this repo!)

File: sphinxarg/ext.py [hunk 2]
```diff
@@ -431,6 +432,31 @@ def _nested_parse_paragraph(self, text):
...
         return content
 
+    def _open_filename(self):
+        # try open with given path
+        try:
+            f = open(self.options['filename'])
+        except (OSError, FileNotFoundError):
+            pass
+        else:
+            return f
```
[the remark this answers is not in the document]
Comment (tsutterley) [reply sphinxarg/ext.py]: hey @ashb no worries on the timing. I think I have all your requested changes in with 5bf2fc6

**Review by ashb — APPROVED** [review]

File: sphinxarg/ext.py [hunk 3]
```diff
@@ -431,6 +432,25 @@ def _nested_parse_paragraph(self, text):
         self.state.nested_parse(StringList(text.split("\n")), 0, content)
         return content
 
+    def _open_filename(self):
+        # try open with given path
+        try:
+            return open(self.options['filename'])
+        except (OSError, FileNotFoundError):
```
Comment (ashb) [inline sphinxarg/ext.py:439]:
```suggestion
        except OSError:
```

File: sphinxarg/ext.py [hunk 4]
```diff
@@ -431,6 +432,25 @@ def _nested_parse_paragraph(self, text):
...
+            pass
+        # try open with abspath
+        try:
+            return open(os.path.abspath(self.options['filename']))
+        except (OSError, FileNotFoundError):
```
Comment (ashb) [inline sphinxarg/ext.py:444]:
```suggestion
        except OSError:
```

File: sphinxarg/ext.py [hunk 5]
```diff
@@ -431,6 +432,25 @@ def _nested_parse_paragraph(self, text):
...
+            pass
+        # try open with shutil which
+        try:
+            return open(shutil.which(self.options['filename']))
+        except (OSError, FileNotFoundError):
```
Comment (ashb) [inline sphinxarg/ext.py:449]:
```suggestion
        except OSError:
```

File: sphinxarg/ext.py [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import os
 import sys
 from argparse import ArgumentParser
+import shutil
```
Comment (ashb) [inline sphinxarg/ext.py:4]:
```suggestion
import shutil
import sys
from argparse import ArgumentParser
```

**Comment by tsutterley** [comment]
thanks for putting in the final touches @ashb!

**Comment by ashb** [comment]
@tsutterley Not a problem, I'll get a release out today/tomorrow with this in. (And a bump to py3.7 as min requirement too.)

**Comment by ashb** [comment]
@tsutterley released as 0.4.0

**Pull request merged**